test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -109,310 +113,386 @@ func (s *snapshottableTestSuite) DefineTests(driver TestDriver, pattern testpatt
-		err = utils.WaitForGVRDeletion(l.dc, SnapshotContentGVR, snapshotContentName, 1*time.Second /* poll */, 30*time.Second /* timeout */)
-		framework.ExpectError(err) // should fail deletion check
-
-		// The purpose of this block is to prevent physical snapshotContent leaks.
-		// We must update the SnapshotContent to have Delete Deletion policy,
-		// or else the physical snapshot content will be leaked.
-		ginkgo.By("get the latest copy of volume snapshot content")
-		snapshotContent, err := l.dc.Resource(SnapshotContentGVR).Get(context.TODO(), l.vscontent.GetName(), metav1.GetOptions{})
+		r.Vs, err = dc.Resource(SnapshotGVR).Namespace(r.Vs.GetNamespace()).Create(context.TODO(), r.Vs, metav1.CreateOptions{})
 		framework.ExpectNoError(err)
 
-		ginkgo.By("updating the the SnapshotContent to have Delete Deletion policy")
-		snapshotContent.Object["spec"].(map[string]interface{})["deletionPolicy"] = "Delete"
-		l.vscontent, err = l.dc.Resource(SnapshotContentGVR).Update(context.TODO(), snapshotContent, metav1.UpdateOptions{})
+		err = WaitForSnapshotReady(dc, r.Vs.GetNamespace(), r.Vs.GetName(), framework.Poll, framework.SnapshotCreateTimeout)
 		framework.ExpectNoError(err)
 
-		ginkgo.By("manually deleting the SnapshotContent")
-		err = l.dc.Resource(SnapshotContentGVR).Delete(context.TODO(), snapshotContent.GetName(), metav1.DeleteOptions{})
-		if err != nil && !apierrors.IsNotFound(err) {
-			framework.Failf("Error deleting snapshot content %q. Error: %v", snapshotContent.GetName(), err)
-		}
+		r.Vs, err = dc.Resource(SnapshotGVR).Namespace(r.Vs.GetNamespace()).Get(context.TODO(), r.Vs.GetName(), metav1.GetOptions{})
+
+		snapshotStatus := r.Vs.Object["status"].(map[string]interface{})
+		snapshotContentName := snapshotStatus["boundVolumeSnapshotContentName"].(string)
+		framework.Logf("received snapshotStatus %v", snapshotStatus)
+		framework.Logf("snapshotContentName %s", snapshotContentName)
+		framework.ExpectNoError(err)
 
-		ginkgo.By("checking the SnapshotContent has been deleted")
-		err = utils.WaitForGVRDeletion(l.dc, SnapshotContentGVR, snapshotContentName, framework.Poll, framework.SnapshotDeleteTimeout)
+		r.Vscontent, err = dc.Resource(SnapshotContentGVR).Get(context.TODO(), snapshotContentName, metav1.GetOptions{})
 		framework.ExpectNoError(err)
-	} else {
-		framework.Failf("Invalid test config. DeletionPolicy should be either Delete or Retain. DeletionPolicy: %v", sct.DeletionPolicy)
+	case testpatterns.PreprovisionedCreatedSnapshot:
+		// prepare a pre-provisioned VolumeSnapshotContent with certain data
+		// Because this could be run with an external CSI driver, we have no way
+		// to pre-provision the snapshot as we normally would using their API.
+		// We instead dynamically take a snapshot and create another snapshot using
+		// the first snapshot's snapshot handle.
+		ginkgo.Skip("Preprovisioned test not implemented")
+		ginkgo.By("taking a snapshot with deletion policy retain")
```

Comment:
Don't think these are needed. "not implemented" is good enough. We can add details when the test is added.

Reply:
True these aren't needed, but it's a scaffolding to remember the steps.